packages/lesswrong/components/users/Leaderboard.tsx

```diff
@@ -0,0 +1,317 @@
+    },
+    "& th:nth-child(1), & td:nth-child(1)": {
+      width: "10%",
+    },
+    "& th:nth-child(2), & td:nth-child(2)": {
+      width: "60%",
+    },
+    "& th:nth-child(3), & td:nth-child(3)": {
+      width: "30%",
+    },
+  },
+}));
+
+type NetKarmaChangesResult = {
+  NetKarmaChangesForAuthorsOverPeriod: { userId: string; netKarma: number }[];
+};
+
+type AirtableLeaderboardResult = {
+  name: string;
+  leaderboardAmount?: number;
+};
+
+// Query for the donor (Airtable) leaderboard
+const DONOR_LEADERBOARD_QUERY = gql`
+  query AirtableLeaderboards {
+    AirtableLeaderboards {
+      name
+      leaderboardAmount
+    }
+  }
+`;
+
+const LeaderboardTable = ({ headers, children }: { headers: string[], children: React.ReactNode }) => {
+  const classes = useStyles(styles);
+  return (
+    <table className={classes.table}>
+      <thead>
+        <tr>
+          {headers.map(header => (
+            <th key={header}>{header}</th>
+          ))}
+        </tr>
+      </thead>
+      {children}
+    </table>
+  );
+};
+
+type KarmaEntry = {
+  userId: string;
+  netKarma: number;
+};
+
+const Leaderboard = () => {
+  const { ErrorBoundary, Loading, UsersNameDisplay, UsersName, LoadMore } = Components;
+  const classes = useStyles(styles);
+  const [timeframe, setTimeframe] = useState<"allTime" | "30days">("allTime");
+
+  // Query #1: Top Karma Users
+  //  - We'll load 50 by default now.
+  const {
+    results: topKarmaUsers,
+    loadMoreProps,
+  } = useMulti({
+    collectionName: "Users",
+    fragmentName: "UsersMinimumInfo",
+    terms: { view: "usersTopKarma" },
+    itemsPerPage: 50,
+    enableTotal: true,
+    limit: 50,
+  });
+
+  // Query #2: Karma changes in last 30 days (also load 50 by default)
+  const { data: last30DaysData, fetchMore, networkStatus } =
+    useQuery<NetKarmaChangesResult>(
+      gql`
+        query getNetKarmaChangesForAuthorsOverPeriod($days: Int!, $limit: Int!) {
+          NetKarmaChangesForAuthorsOverPeriod(days: $days, limit: $limit) {
+            userId
+            netKarma
+          }
+        }
+      `,
+      {
+        variables: {
+          days: 30,
+          limit: 50,
+        },
+      }
+    );
+
+  // Query #3: Donor leaderboard (from Airtable) -- load 50 donors to match the user karma query's default.
+  const { data: donorData, loading: donorLoading, error: donorError } =
+    useQuery<{ AirtableLeaderboards: AirtableLeaderboardResult[] }>(
+      DONOR_LEADERBOARD_QUERY
+    );
+
+  // We'll separate donors into those with a known numeric donation, and those over 5k with no exact data
+  const donors = donorData?.AirtableLeaderboards || [];
+  const donorsWithAmounts = donors.filter(
+    (entry) => typeof entry.leaderboardAmount === "number"
+  );
+  const donorsOver5kUnknown = donors.filter(
+    (entry) => typeof entry.leaderboardAmount !== "number"
+  );
+
+  const last30DaysResults =
+    last30DaysData?.NetKarmaChangesForAuthorsOverPeriod || [];
```

Comment:
(won't go through all of them, but nit: not long enough to need a line break.  also, prefer `??` to `||` when possible)